### PR TITLE
 Implement getBLSKeyPairFromPassphraseAndPath for BLS key derivation - Closes #7196 

### LINKED ISSUE #7196: Implement `getBLSKeyPairFromPassphraseAndPath` for BLS key derivation
### Description

This function derives a BLS key pair, as defined in draft-irtf-cfrg-bls-signature-04, from an input phrase and a derivation path.

```
getBLSPrivateKeyFromPhraseAndPath(phrase: bip39Mnemonic, path: list[uint32]):
    masterSeed = bip39.toSeed(phrase)
    key = derive_master_SK(masterSeed) // use `bls.keyGen from lisk-crypto`


    for index in path:
        key = derive_child_SK(key, index)
    return key
```

### Acceptance Criteria
- Add unit tests

### PR DESCRIPTION
**Pull request by Incede**
### What was the problem?

This PR resolves #7196 

### How was it solved?

Implement the function as specified

### How was it tested?

Added unit test

### CODE REVIEW

File: elements/lisk-cryptography/src/utils.ts [hunk 1]
```diff
@@ -93,3 +94,90 @@ export const getChildKey = (node: { key: Buffer; chainCode: Buffer }, index: num
...
+
+// eslint-disable-next-line no-bitwise
+const flipBits = (buf: Buffer) => Buffer.from(buf.map(x => x ^ 0xff));
+
+const sha256 = (x: Buffer) => crypto.createHash('sha256').update(x).digest();
+
+const hkdfSHA256 = (ikm: Buffer, length: number, salt: Buffer, info: Buffer) => {
+	if (salt.length === 0) {
+		// eslint-disable-next-line no-param-reassign
+		salt = Buffer.from([
```
Comment (ishantiw) [inline elements/lisk-cryptography/src/utils.ts:106]: maybe we can declare a constant for this?

File: elements/lisk-cryptography/src/utils.ts [hunk 2]
```diff
@@ -93,3 +94,90 @@ export const getChildKey = (node: { key: Buffer; chainCode: Buffer }, index: num
...
+			0,
+			0,
+			0,
+			0,
+			0,
+			0,
+			0,
+		]);
+	}
+	const PRK = crypto.createHmac('sha256', salt).update(ikm).digest();
```
Comment (ishantiw) [inline elements/lisk-cryptography/src/utils.ts:141]: we can create a util function maybe similar to code sample
```ts
const hmacSHA256 = (key, message, hash) => {
    return createHmac(hash, key).update(message).digest()
}
```

File: elements/lisk-cryptography/src/utils.ts [hunk 3]
```diff
@@ -93,3 +94,90 @@ export const getChildKey = (node: { key: Buffer; chainCode: Buffer }, index: num
...
+			.update(Buffer.concat([t, info, Buffer.from([1 + i])]))
+			.digest();
+		OKM = Buffer.concat([OKM, t]);
+	}
+	return OKM.slice(0, length);
+};
+
+const toLamportSK = (IKM: Buffer, salt: Buffer) => {
+	const K = 32;
+	const L = K * 255;
```
Comment (ishantiw) [inline elements/lisk-cryptography/src/utils.ts:157]: maybe these can be declared as constants too?

**Review by ishantiw — COMMENTED** [review]
LGTM, just a few suggestions

File: elements/lisk-cryptography/src/constants.ts [hunk 4]
```diff
@@ -20,3 +20,38 @@ export const ED25519_CURVE = 'ed25519 seed';
 export const MAX_UINT32 = 4294967295;
 export const HARDENED_OFFSET = 0x80000000;
 export const HASH_LENGTH = 32;
+export const L = 8160; // 32 * 255
+export const EMPTY_SALT = Buffer.from([
```
Comment (ishantiw) [inline elements/lisk-cryptography/src/constants.ts:24]: maybe we can use something like,
`export const EMPTY_SALT= Buffer.from(new Array(32).fill(0))`

File: elements/lisk-cryptography/src/utils.ts [hunk 5]
```diff
@@ -100,61 +107,26 @@ const flipBits = (buf: Buffer) => Buffer.from(buf.map(x => x ^ 0xff));
...
-			0,
-			0,
-			0,
-			0,
-			0,
-		]);
+		salt = EMPTY_SALT;
 	}
-	const PRK = crypto.createHmac('sha256', salt).update(ikm).digest();
+	const PRK = hmacSHA256(salt, ikm, 'sha256');
```
Comment (ishantiw) [inline elements/lisk-cryptography/src/utils.ts:118]: `export const SHA256 = 'sha256'` could be a constant too

**Review by ishantiw — COMMENTED** [review]
Overall LGTM, couple of minor suggestions

**Review by mitsuaki-u — APPROVED** [review]
Great work!!

**Review by ishantiw — APPROVED** [review]
Great work 🎉

**Pull request merged**